packages/perseus/src/util/graphie-utils.test.ts

```diff
@@ -0,0 +1,43 @@
+import {parseDataFromJSONP} from "./graphie-utils";
+import {typicalCase, edgeCases} from "./graphie-utils.testdata";
+
+describe("graphie utils", () => {
+    const errorCallback = jest.fn((error) => {
+        // Do nothing
+    });
+
+    it("should parse the modern graphie json format", () => {
+        // Act
+        const data = parseDataFromJSONP(
+            typicalCase.hash,
+            typicalCase.jsonpString,
+            errorCallback,
+        );
+
+        // Assert
+        expect(data).toEqual(typicalCase.expectedData);
+    });
+
+    it("should call the error callback when unable to parse JSONP", () => {
+        // Act
+        parseDataFromJSONP(typicalCase.hash, "invalid jsonp", errorCallback);
+
+        // Assert
+        expect(errorCallback).toHaveBeenCalled();
+    });
+
+    // also test the array of edge cases from the testdata file
+    edgeCases.forEach((edgeCase, i) => {
+        it(`should parse the edge case ${i}`, () => {
```

Comment:
There's a lot of edge cases, and some of them are quite similar. I could reduce them down, potentially, but I did like that we could cover 99% of the cases outlined in the ticket as failing to parse. Happy to take feedback on this one!

Reply:
I think the collection of edge cases you have is probably good. The ones that will fail (like that one with a `u''` prefix on all strings, is likely broken in prod too so I'd guess it would need to be re-written to work anyways.